### PR TITLE
PA-4170 | Fixing build and implementing gradle build

### PR DESCRIPTION
**Pull request by AbhishekRatnawat**
PR is ready, we are awaiting JFrog-Cli to get enabled on pipeline. 

Blocking ticket: https://acquia.atlassian.net/browse/PT-10057

### CODE REVIEW

File: .acquia/Dockerfile.ci [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM public.ecr.aws/amazoncorretto/amazoncorretto:11-al2023-jdk
+FROM public.ecr.aws/amazoncorretto/amazoncorretto:11-al2022-jdk
```
Comment (AbhishekRatnawat) [inline .acquia/Dockerfile.ci:1]:
```suggestion
FROM public.ecr.aws/amazoncorretto/amazoncorretto:11-al2023-jdk
```

File: build.gradle [hunk 2]
```diff
@@ -62,4 +67,4 @@ subprojects {
     java {
         withSourcesJar()
     }
-}
+}
```
Comment (amanjoshicodes) [inline build.gradle:70]: Please add new line at EOF.

File: build.gradle [hunk 3]
```diff
@@ -4,41 +4,46 @@ subprojects {
     apply plugin: 'maven-publish'
 
     group = 'widen'
-    version = '2.1.0-SNAPSHOT'
+    // version = 2.1.0-SNAPSHOT
```
Comment (amanjoshicodes) [inline build.gradle:7]: Clean it up and Provide link to Java based repo onboarding doc that we have created as a comment here.

File: .acquia/pipeline.yaml [hunk 4]
```diff
@@ -27,8 +27,20 @@ pre_build:
...
+        - "2.1.0-SNAPSHOT"
       props:
         - "maintainer=widen"
-
+security_scan:
+  scanner: orca
+  ignore_failures: true
+  reason: "This is an internal/testing/no-production application. The vulnerabilities would be fixed by 2023/12/31 in ticketnumber PA-3838"
+  
+deploy:
```
Comment (amanjoshicodes) [inline .acquia/pipeline.yaml:41]: Why are we encouraging `custom` deployment here?

File: .acquia/pipeline.yaml [hunk 5]
```diff
@@ -27,8 +27,20 @@ pre_build:
...
+      build_tool: "gradlew"
+      custom_version:
+        - "2.1.0-SNAPSHOT"
       props:
         - "maintainer=widen"
-
+security_scan:
+  scanner: orca
+  ignore_failures: true
+  reason: "This is an internal/testing/no-production application. The vulnerabilities would be fixed by 2023/12/31 in ticketnumber PA-3838"
```
Comment (amanjoshicodes) [inline .acquia/pipeline.yaml:39]: Please update the date, I'm skeptical that we would be able to address it by `2023/12/31`

**Review by amanjoshicodes — CHANGES_REQUESTED** [review]
Minor suggestions, Please add relevant screenshots and update PR description.

File: .acquia/pipeline.yaml (hunk 4 above)
Comment (AbhishekRatnawat) [reply .acquia/pipeline.yaml:41]: This application is not deploying anything. Included deploy section just to make package promotion work.

File: .acquia/pipeline.yaml (hunk 5 above)
Comment (AbhishekRatnawat) [inline .acquia/pipeline.yaml:39]:
```suggestion
  reason: "This is an internal/testing/no-production application. The vulnerabilities would be fixed by 2024/03/31 in ticketnumber PA-3838"
```

File: build.gradle [hunk 6]
```diff
@@ -4,41 +4,46 @@ subprojects {
...
+        //         url = 'https://jfrog.ais.acquia.io/artifactory/libs-snapshots-local'
+        //         credentials(HttpHeaderCredentials) {
+        //             name = 'Authorization'
+        //             value = "Bearer " + (System.env.ACQUIA_JFROG_TOKEN ?: ACQUIA_JFROG_TOKEN)
+        //         }
+        //         authentication {
+        //             header(HttpHeaderAuthentication)
+        //         }
+        //     }
+        // }
```
Comment (AbhishekRatnawat) [inline build.gradle:38]:
```suggestion
```

File: build.gradle [hunk 7]
```diff
@@ -4,41 +4,46 @@ subprojects {
...
+              username = System.env.ACQUIA_JFROG_USER
+              password = System.env.ACQUIA_JFROG_TOKEN
+          }
+        // credentials(HttpHeaderCredentials) {
+        //             name = 'Authorization'
+        //             value = "Bearer " + (System.env.JFROG_API_TOKEN ?: JFROG_API_TOKEN)
+        //         }
+        //         authentication {
+        //             header(HttpHeaderAuthentication)
+        //         }
```
Comment (AbhishekRatnawat) [inline build.gradle:22]:
```suggestion
```

File: build.gradle (hunk 3 above)
Comment (AbhishekRatnawat) [inline build.gradle:7]:
```suggestion
```

File: build.gradle (hunk 2 above)
Comment (AbhishekRatnawat) [inline build.gradle:50]:
```suggestion
}

```

File: build.gradle [hunk 8]
```diff
@@ -4,41 +4,26 @@ subprojects {
     apply plugin: 'maven-publish'
 
     group = 'widen'
```
Comment (AbhishekRatnawat) [inline build.gradle:6]:
```suggestion
// refer to: https://github.com/acquia/devops-pipeline/tree/master/consumer/java_projects_configurations.md to understand updated configurations in this file.
    group = 'widen'
```

File: build.gradle [hunk 9]
```diff
@@ -3,42 +3,28 @@ subprojects {
     apply plugin: 'java'
     apply plugin: 'maven-publish'
 
+// refer to: https://github.com/acquia/devops-pipeline/tree/master/consumer/java_projects_configurations.md to understand updated configurations in this file.
```
Comment (AbhishekRatnawat) [inline build.gradle:6]: will update this link once the documentation is merged.

File: .acquia/pipeline.yaml (hunk 5 above)
Comment (AbhishekRatnawat) [reply .acquia/pipeline.yaml]: done

File: build.gradle (hunk 3 above)
Comment (AbhishekRatnawat) [reply build.gradle]: done

File: build.gradle (hunk 2 above)
Comment (AbhishekRatnawat) [reply build.gradle]: done

**Review by amanjoshicodes — APPROVED** [review]
LGTM

**Pull request merged**